02-api-reference/01-agent-frameworks/02-python/02-vision-agent-api/locators.mdx

```diff
@@ -0,0 +1,1125 @@
+
+- if most of **A** (or, more specifically, **A**'s bounding box) is *right of* **B**
+(or, more specifically, the **right border** of **B**'s bounding box) **and**
+- if the **left border** of **A** (or, more specifically, **A**'s bounding box) is
+*right of* the **left border** of **B** (or, more specifically, **B**'s
+bounding box).
+
+**Arguments**:
+
+- `other_locator` _Relatable_ - Locator for an element / elements to relate to
+- `index` _RelationIndex, optional_ - Index of the element (located by *self*) **right of** the other
+  element(s) (located by *other_locator*), e.g., the first (`0`), second (`1`), third (`2`) etc. element right of the other
+  element(s). Elements' (relative) position is determined by the **left
+  border** (*x*-coordinate) of their bounding box.
+  We don't guarantee the order of elements with the same left border
+  (*x*-coordinate). Defaults to `0`.
+- `reference_point` _ReferencePoint, optional_ - Defines which element (located by *self*) is considered to be
+  *right of* the other element(s) (located by *other_locator*). Defaults to `"center"`.
+  
+
+**Returns**:
+
+- `Self` - The locator with the relation added
+  
+
+**Examples**:
+
+    ```text
+
+    =========== ===========
+    |    B    | |    A    |
+    =========== ===========
+    ```
+    ```python
+    from askui import locators as loc
+    # locates text "A" as it is the first (index 0) element right of ("center"
+    # of) text "B"
+    text = loc.Text().right_of(loc.Text("B"), reference_point="center")
+    ```
+  
+    ```text
+
+    =========== 
+    |    B    | 
+    =========== ===========
+                |    A    |
+                ===========
+    ```
+    ```python
+    from askui import locators as loc
+    # locates text "A" as it is the first (index 0) element right of 
+    # ("boundary" of / any point of) text "B" 
+    # (reference point "center" won't work here)
+    text = loc.Text().right_of(loc.Text("B"), reference_point="boundary")
+    ```
+  
+    ```text
+
+    =========== 
+    |    B    |
+    ===========
+                ===========
+                |    A    |
+                ===========
+    ```
+    ```python
+    from askui import locators as loc
+    # locates text "A" as it is the first (index 0) element right of text "B" 
+    # (reference point "center" or "boundary" won't work here)
+    text = loc.Text().right_of(loc.Text("B"), reference_point="any")
+    ```
+  
+    ```text
+
+                            ===========
+                            |    A    |
+                            ===========
+    =========== ===========
+    |    C    | |    B    |
+    =========== ===========
+    ```
+    ```python
+    from askui import locators as loc
+    # locates text "A" as it is the second (index 1) element right of text "C" 
+    # (reference point "center" or "boundary" won't work here)
+    text = loc.Text().right_of(loc.Text("C"), index=1, reference_point="any")
```

Comment:
Is this right? Customer Reported that this is not working: https://askyourui.atlassian.net/browse/CL-1370

Reply:
There was a bug in serialization of `Text()` without actual content. Fixed in [v0.3.2](https://github.com/askui/vision-agent/releases/tag/v0.3.2).